src/backend/cuda/platform.cpp

```diff
@@ -332,10 +342,10 @@ DeviceManager::DeviceManager()
 
     sortDevices();
 
-    for(int i = 0; i < nDevices; i++) {
-        setActiveDevice(i, cuDevices[i].nativeId);
-        CUDA_CHECK(cudaStreamCreate(&streams[i]));
-    }
+    // Initialize all streams to 0.
+    // Streams will be created in setActiveDevice()
+    for(int i = 0; i < (int)MAX_DEVICES; i++)
+        streams[i] = (cudaStream_t)0;
```

Comment:
I understand what it is doing, which is exactly what i avoided it in first place, but it had a bug i didn't notice that got fixed now.  

Init time is not a concern in many cases, since it is going to be a little higher whether this particular call is there in the initialization or not. So, doing it in the beginning is better than doing later which makes `setDevice` a very quick switch.